static/app/utils/marked/marked.tsx

```diff
@@ -0,0 +1,144 @@
+import dompurify from 'dompurify';
+import type {Tokens} from 'marked'; // eslint-disable-line no-restricted-imports
+import {Marked, marked} from 'marked'; // eslint-disable-line no-restricted-imports
+import {markedHighlight} from 'marked-highlight';
+import Prism from 'prismjs';
+
+import {loadPrismLanguage} from 'sentry/utils/prism';
+
+// Only https and mailto, (e.g. no javascript, vbscript, data protocols)
+const safeLinkPattern = /^(https?:|mailto:)/i;
+
+const safeImagePattern = /^https?:\/\/./i;
+
+function isSafeHref(href: string, pattern: RegExp) {
+  try {
+    return pattern.test(decodeURIComponent(unescape(href)));
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Implementation of marked. Renderer which additionally sanitizes URLs.
+ */
+class SafeRenderer extends marked.Renderer {
+  link({href, title, text, ...rest}: Tokens.Link) {
+    // For a bad link, just return the plain text href
+    if (!isSafeHref(href, safeLinkPattern)) {
+      return href;
+    }
+
+    const out = super.link({href, title, text, ...rest});
+    return dompurify.sanitize(out);
+  }
+
+  image({href, title, text, ...rest}: Tokens.Image) {
+    // For a bad image, return an empty string
+    if (!isSafeHref(href, safeImagePattern)) {
+      return '';
+    }
+
+    return super.image({href, title, text, ...rest});
+  }
+}
+
+class NoParagraphRenderer extends SafeRenderer {
+  paragraph(tokens: Tokens.Paragraph) {
+    // Do not render the paragraph but still render sub-tokens
+    return super.text({...tokens, type: 'text'});
+  }
+}
+
+function preprocess(markdown: string) {
+  // Allow no html tags at all in the preprocess step.
+  // GitHub and others allow this, we could in the future.
+  return dompurify.sanitize(markdown, {ALLOWED_TAGS: []});
+}
+
+function postprocess(html: string) {
+  return dompurify.sanitize(html);
+}
+
+const noHighlightingMarked = new Marked({
+  async: false,
+  renderer: new SafeRenderer(),
+  hooks: {
+    preprocess,
+    postprocess,
+  },
+});
+
+const highlightingMarked = new Marked(
+  markedHighlight({
+    async: true,
```

Comment:
this is the main reason for the changes, to support syntax highlighting marked has to return a Promise